lib/spack/spack/environment/environment.py

```diff
@@ -1475,7 +1475,10 @@ def _concretize_separately(self, tests=False):
 
         # Solve the environment in parallel on Linux
         start = time.time()
-        max_processes = min(len(arguments), 16)  # Number of specs  # Cap on 16 cores
+        max_processes = min(
+            len(arguments),  # Number of specs
+            spack.config.get("config:build_jobs"),  # Cap on build jobs
```

Comment:
Are you sure we should still be taking the min of the two arguments here? If a user declares `-j 16` but only specifies a single spec they might be surprised to see clingo only using a single thread no? Or is this not controlling threads to clingo?

Reply:
`clingo` is currently single threaded. We checked in the early days and using parallelism at that level was detrimental to performance. Parallelism is obtained by spawning multiple processes, each of which is concretizing an independent spec.

Reply:
Reference for using a single thread:

https://github.com/spack/spack/blob/384f5f99605526b09de694729daa4eb759ed03b8/lib/spack/spack/solver/asp.py#L83

Reply:
Corollary to this is if we switch to `when_possible` in our CI environments, we're back to basically one thread. IMO this is fine, as it'll still (hopefully) be faster b/c it's a single solve. But the win with parallelism here is b/c we're concretizing so many things separately.

Reply:
> `clingo` is currently single threaded. We checked in the early days and using parallelism at that level was detrimental to performance. Parallelism is obtained by spawning multiple processes, each of which is concretizing an independent spec.

Gotcha! Thanks for the explanation @alalazo! I was definitely confused on the meaning of processes here.